service/src/main/java/com/bankbazaar/kafka/service/config/KafkaStreamsConfig.java

```diff
@@ -0,0 +1,151 @@
+        return retryTemplate;
+    }
+    /**
+     * Set status to IN_PROGRESS.
+     * Consumes data from Create_CSV topic.
+     * Validates if file already exits.
+     * Set status to FAILURE if file already exists.
+     * Output data to File_Processor topic.
+     */
+    @Bean
+    public Function<KStream<String,Data>, KStream<String,Data>> fileProcessor(@Qualifier("streamRetryTemplate") RetryTemplate retryTemplate)
+    {
+        return kStream -> kStream.filter((key, value) ->
+                {
+                    return retryTemplate.execute(
+                            retryContext -> {
+                                try {
+                                    FileStatusEntity fileData = new FileStatusEntity();
+                                    fileData.setId(value.getId());
+                                    fileData.setStatus(Status.IN_PROGRESS);
+                                    fileStatusService.update(fileData);
+                                    ClassLoader classLoader = getClass().getClassLoader();
+                                    File file = new File(classLoader.getResource(".").getFile() + value.getFileName());
+                                    if (file.exists()) {
+                                        fileData.setStatus(Status.FAILURE);
+                                        fileStatusService.update(fileData);
+                                        return false;
+                                    }
+                                    return true;
+                                }
+                                catch (Exception exception)
+                                {
+                                    log.error("retrying",exception);
+                                    throw exception;
+                                }
+                                },
+
+                            context -> {
+                                log.error("retries exhausted");
+                                FileStatusEntity fileData = new FileStatusEntity();
+                                fileData.setId(value.getId());
+                                fileData.setStatus(Status.ERROR);
+                                fileStatusService.update(fileData);
+                                return false;
+                            }
+                            );
+                }
+                );
+    }
+
+    /**
+     * Consumes data from File_Processor topic.
+     * Create.csv file and populate file using data.
+     * Set status to SUCCESS if file IS created successfully.
+     * Set status to ERROR if exception occurs during file creation.
+     * Retry a maximum of 1 time if exception occurs;
+     */
+    @Bean
+    public Consumer<KStream<String,Data>> consumer(@Qualifier("streamRetryTemplate") RetryTemplate retryTemplate)
+    {
+        return kStream -> kStream.foreach((key, value) ->
+                {
+                    try {
```

Comment:
The Retry template + context itself is like a try catch. No need to add it explicitly

Reply:
this has been removed